parser.rb

```diff
@@ -0,0 +1,43 @@
+
+class Parser
+  def initialize(filename)
+    @file = File.open(filename, "r")
+    @text = ''
+    @file.each_line do |txt|
+      @text = @text + txt
+    end
+  end
+
+  def array_of_words
+    @text.split(/\W+/)
+  end
+
+  def map_words_to_count
+    array = array_of_words
+    obj = {}
+    array.each do |item|
+      item = item.downcase
+      obj[item] = obj[item] + 1 if obj[item] != nil
+      obj[item] = 1 if obj[item] == nil
+    end
+    obj
+  end
+
+  def decreasing_order_word_count
+    map = map_words_to_count
+    out_array = []
+    map.each do |item|
+      new_obj = {name: item[0], value: item[1]}
+      out_array.push new_obj
+    end
+    out_array.sort_by {|item| item[:value]}.reverse
+  end
```

Comment:
whoa.  This doesn't need to be so complicated.  Try `.sort_by { |_key, value| value }` on a hash.